ide/src/tests/slice/select.test.ts

```diff
@@ -0,0 +1,28 @@
+import chai, { expect } from "chai";
+import deepEqualAnyOrder from 'deep-equal-in-any-order';
+import { suite, before, test } from "mocha";
+import _ from "lodash";
+import { forward_slices, backward_slices } from "./mock_data/slices";
+import { get_slice_selections, resolve_sequentially } from "./util/slice_helpers";
+
+suite("Slice selection tests", async () => {
+  before(() => {
+    chai.use(deepEqualAnyOrder);
+  });
+
+  test("forward select", async () => {
+    const forward_selections = await resolve_sequentially(forward_slices, get_slice_selections);
+
+    forward_selections.forEach(async (selection) => {
+      expect(selection.expected_selections).to.be.deep.equalInAnyOrder(selection.actual_selections);
+    });
+  });
+
+  test("backward select", async () => {
+    const backward_selections = await resolve_sequentially(backward_slices, get_slice_selections);
+
+    backward_selections.forEach(async (selection) => {
+      expect(selection.expected_selections).to.be.deep.equalInAnyOrder(selection.actual_selections);
+    });
+  });
```

Comment:
I split the [previous method](https://github.com/willcrichton/flowistry/blob/5c805f25142b8d27bfd069601709c477657ba5d1/ide/src/tests/unit/extension.test.ts#L94-L103) for testing the slice selections into two separate `test`s because this seems more readable and removes the code-smell of having to create a useless `it()`.

Let me know if you preferred the previous method though, since it did reduce code duplication.

Reply:
Yeah since the code is identical I would still factor out a function. You could use a curried function to make it ✨ beautiful ✨ 

```js
const slice_test = slices => async () => {
  const selections = await resolve_sequentially(slices, get_slice_selections);
  selections.forEach(...)
}

test("forward select", slice_test(forward_slices));
```